osquery/remote/plugins/remote_enrollment.cpp

```diff
@@ -0,0 +1,94 @@
+#include <osquery/enrollment.h>
+
+#include "osquery/remote/requests.h"
+#include "osquery/remote/transports/http.h"
+#include "osquery/remote/serializers/json.h"
+
+#define MAX_TRIES 5
+
+namespace osquery {
+
+FLAG(string,
+     enrollment_endpoint,
+     "Not Specified",
+     "The endpoint for server side client enrollment");
+
+FLAG(string,
+     enrollment_port,
+     "Not Specified",
+     "The endpoint's port for server side client enrollment");
```

Comment:
instead of host and port, you should just make a uri flag. cpp-netlib will handle parsing them out when you create an osquery::Request object.